http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/DevModeWorkspaceIT.java

```diff
@@ -0,0 +1,101 @@
+
+import java.nio.file.Path;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import com.microsoft.playwright.Browser;
+import com.microsoft.playwright.ElementHandle;
+import com.microsoft.playwright.Page;
+import com.microsoft.playwright.Playwright;
+
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.DevModeQuarkusApplication;
+import io.quarkus.test.utils.FileUtils;
+
+@Tag("QUARKUS-6247")
+@QuarkusScenario
+public class DevModeWorkspaceIT {
+
+    @DevModeQuarkusApplication(properties = "keycloakless.properties")
+    static final RestService app = new RestService();
+
+    @Test
+    public void workSpaceExists() {
+        app.given().when().get("/q/dev-ui/workspace").then().statusCode(HttpStatus.SC_OK);
+    }
+
+    @Test
+    public void workSpaceContainsText() {
+        var pageURL = app.getURI(Protocol.HTTP).withPath("/q/dev-ui/workspace").toString();
+        try (Playwright playwright = Playwright.create()) {
+            try (Browser browser = playwright.firefox().launch()) {
+                Page page = browser.newContext().newPage();
+                page.navigate(pageURL);
+                ElementHandle element = page.waitForSelector("#code");
+                String code = element.getAttribute("value");
+                Assertions.assertTrue(code.startsWith("package io.quarkus.ts.http.advanced;"),
+                        "The code doesn't contain the expected value: " + code);
+            }
+        }
+    }
+
+    @Test
+    public void workSpaceCanBeEdited() {
+        // Check, that the file is in original state, and query parameter "names" doesn't exist
+        app.given().when().get("/api/hello?name=Editor")
+                .then().statusCode(HttpStatus.SC_OK).body(containsString("Hello, Editor!"));
+        app.given().when().get("/api/hello?names=Editors")
+                .then().statusCode(HttpStatus.SC_OK).body(containsString("Hello, World!"));
+
+        var pageURL = app.getURI(Protocol.HTTP).withPath("/q/dev-ui/workspace").toString();
+        try (Playwright playwright = Playwright.create()) {
+            try (Browser browser = playwright.firefox().launch()) {
+                Page page = browser.newContext().newPage();
+                page.navigate(pageURL);
+
+                // switch to HelloResource.java
+                ElementHandle treeSelector = page.waitForSelector(
+                        "ul.tree:nth-child(1) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(1) > ul:nth-child(2) > li:nth-child(14) > div:nth-child(1) > span:nth-child(2)");
```

Comment:
I want to look at this, if there is some more readable selector as from my pov this could be changed quite often.

Reply:
@fedinskiy please talk to @phillip-kruger about that. The current selector is too fragile, and there is a high probability that it will fail with any refactoring in this area.

Reply:
You need to add ID, I am surprised that you could locate anything in the shadow root by text @jedla97 , really it goes directly against my experience with this.

Reply:
@michalvavrik looking at it and it's in shadow root and vaadin should be used.
The full xpath `/html/body/div[1]/div/qwc-workspace//div/vaadin-split-layout/master-content/qui-directory-tree//ul/li[1]/ul/li/ul/li[1]/ul/li/ul/li/ul/li/ul/li/ul/li/ul/li[14]/div/span[2]/text()` 

In general I start with assumption that it can start in list.

For Id if it's enough to have id outside of shadow root is should be fine otherwise there will be need for some update on Quarkus side as the list don't have any element with id. There is `qui-directory-tree (id=directory-tree) -> shadow root -> ul`

Reply:
Yeah adding id should be easy enough.

Reply:
Refactored test to avoid using the selector. Also, fixed a data race, which was in the original version of the code